sdk/core/azure-core-amqp/src/models/amqp_properties.cpp

```diff
@@ -296,20 +296,9 @@ namespace Azure { namespace Core { namespace Amqp { namespace Models {
     std::string timeToString(std::chrono::system_clock::time_point t)
     {
       std::time_t time = std::chrono::system_clock::to_time_t(t);
-//#ifdef _MSC_VER
-//#pragma warning(push)
-//// warning C4996: 'ctime': This function or variable may be unsafe. Consider using ctime_s
-//// instead.
-//#pragma warning(disable : 4996)
-//#endif
-      char buf[128] = {0};
-      ctime_s(buf, sizeof(buf), &time);
-      std::string time_str = buf;
-//#ifdef _MSC_VER
-//#pragma warning(pop)
-//#endif
-      time_str.resize(time_str.size() - 1);
-      return time_str;
+      char buf[26]{};
+      std::strftime(buf, std::extent<decltype(buf)>::value, "%c", std::localtime(&time));
```

Comment:
Up to you, but maybe use `std::array`? It has `size()`. And I think it default-initializes.
If you decide to use `std::array`, I've seen another instance of using `char buf[...]` in this PR, I'd recommend to update that one as well.

Reply:
I'm not sure it matters in this case. strftime is documented as taking a char array of size at least 26 bytes. I'm not sure that there is sufficient value in using std::array here.

Also, at least for clang, [the code for the std::array option is significantly more complicated](https://www.godbolt.org/z/TsMdxrsW4).

Reply:
Hmm, I do not see what is different for Clang - looks normal to me.

Reply:
IMHO the generated code in `print_time_array` is dramatically larger and more complicated than the generated code in `print_time_buf`.

I've not checked the visual studio generated code, but I suspect it will be similar.